python/lsst/ts/externalscripts/base_take_twilight_flats.py

```diff
@@ -0,0 +1,457 @@
+
+        Parameters
+        ----------
+        sky_counts : float
+            Counts in electrons of previous flat.
+        exp_time : float
+            Exposure time of previous flat
+
+        Returns
+        -------
+        float
+            Calculated new exposure time.
+        """
+
+        return self.config.target_sky_counts * exp_time / sky_counts
+
+    def get_target_radec(self):
+        """
+        Returns the RADEC of the target area of the sky that's an azimuth
+        `distance_from_sun` away from the Sun, given `elevation`,
+        and at a given `time`.
+
+        Parameters
+        ----------
+        distance_from_sun : float, (-180, 180)
+            The distance from the Sun in degrees.
+            Positive angles go towards the North.
+        elevation : float
+            Target elevation for Sky Flats.
+        """
+
+        az_sun, el_sun = self.tcs.get_sun_azel()
+
+        target_az = (az_sun + self.config.distance_from_sun) % 360
+
+        target_radec = self.tcs.radec_from_azel(target_az, self.config.target_el)
+
+        return target_radec
+
+    def get_empty_field(self, target, radius=5):
+        """
+        Query the "Deep blank field catalogue : J/MNRAS/427/679" in Vizier.
+
+        Parameters
+        ----------
+        target : astropy.coordinates.SkyCoord
+            Sky coordinates near the field
+        radius : float
+            Search radius in degrees.
+
+        Reference
+        ---------
+        http://cdsarc.u-strasbg.fr/viz-bin/Cat?J/MNRAS/427/679
+        """
+        _table = Vizier.query_region(
+            catalog="J/MNRAS/427/679/blank_fld",
+            coordinates=target,
+            radius=radius * u.deg,
+        )
+
+        if len(_table) == 0:
+            self.log.info(
+                f"Could not find a field near {target} " f"within {radius} deg radius"
+            )
+            return None
+
+        _table = _table["J/MNRAS/427/679/blank_fld"]
+
+        coords = coordinates.SkyCoord(
+            ra=_table["RAJ2000"],
+            dec=_table["DEJ2000"],
+            unit=(u.hourangle, u.deg),
+            frame=coordinates.ICRS,
+        )
+
+        arg = target.separation(coords).argmin()
+
+        return coords[arg]
+
+    def assert_sun_location(self):
+        """Confirm sun's elevation is safe for taking twilight flats."""
+        sun_coordinates = self.tcs.get_sun_azel()
+        where_sun = "setting" if (sun_coordinates[0] > 180) else "rising"
+        self.log.debug(
+            f" The azimuth of the {where_sun} Sun is {sun_coordinates[0]:.2f} deg \n"
+            f" The elevation of the Sun is {sun_coordinates[1]:.2f} deg"
+        )
+
+        if (sun_coordinates[1] < self.config.min_sun_elevation) or (
+            sun_coordinates[1] > self.config.max_sun_elevation
+        ):
+            raise AssertionError(
+                f"Sun elevation {sun_coordinates} is outside appropriate elvation limits. Aborting."
```

Comment:
There is a typo in "elvation" -> "elevation". But also, consider removing "Aborting" at the end and also logging what are the conditions. Something like:

```
f"Sun elevation {sun_coordinates[1]} is outside appropriate limits. "
f"Must be below {self.config.min_sun_elevation} or above {self.config.max_sun_elevation}."
```